src/components/hosts/Hostname.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { Button, ButtonVariant } from '@patternfly/react-core';
+import EditHostModal from './EditHostModal';
+import { Host, Inventory, Cluster } from '../../api/types';
+
+type HostnameProps = {
+  host: Host;
+  inventory: Inventory;
+  cluster: Cluster;
+};
+
+export const computeHostname = (host: Host, inventory: Inventory) =>
+  host.requestedHostname || inventory.hostname;
+
+const Hostname: React.FC<HostnameProps> = ({ host, inventory, cluster }) => {
+  const [isOpen, setOpen] = React.useState(false);
+
+  const hostname = computeHostname(host, inventory);
+  const isHostnameChangeRequested = host.requestedHostname !== inventory.hostname;
+
+  return (
+    <>
+      <Button variant={ButtonVariant.link} isInline onClick={() => setOpen(true)}>
+        {hostname}
+        {isHostnameChangeRequested && ' *'}
+      </Button>
+      <EditHostModal
```

Comment:
We're rendering the modal in 2 different places, we could just call `setShowEditHostModal` on the button click if we pass it as a prop (by adding it as a parameter to `hostToHostTableRow` function)

Reply:
I understand your point, but it's a single row which we replace by another single row call but with callback. If it is not a blocker, I prefer to keep it as it is.